packages/ndla-ui/src/MessageBox/MessageBox.tsx

```diff
@@ -93,62 +124,87 @@ const CloseButton = styled(Button)`
+    SongTi, 宋体, sans-serif;
+  ${mq.range({ until: breakpoints.mobileWide })} {
+    margin: 0px 15px 1px 5px;
+    box-shadow: none;
   }
 `;
-
 type Props = {
   type?: WrapperProps['boxType'];
   heading?: string;
   sticky?: boolean;
-  onClose?: () => void;
-  children: React.ReactNode;
+  onClose?: boolean;
+  children?: React.ReactNode;
+  links?: []; //Takes and object with a name and a link. Eks: {'link1', 'www.link.no'}
 };
 
-export const MessageBox = ({
-  heading,
-  type = 'info',
-  sticky = true,
-  onClose,
-  children,
-  t,
-}: Props & WithTranslation) => (
-  <Sticky disabled={!sticky} stickyStyle={{ zIndex: 9999 }}>
-    <Wrapper boxType={type}>
-      <InfoWrapper>
-        <IconWrapper>
-          <InformationOutline style={{ width: '24px', height: '24px' }} />
-        </IconWrapper>
-        <TextWrapper>
-          {heading && <Label>{heading}</Label>}
-          {children}
-        </TextWrapper>
-      </InfoWrapper>
-      {onClose && (
-        <CloseButtonWrapper>
-          <CloseButton link onClick={onClose}>
-            <CloseButtonText>{t('close')}</CloseButtonText>
-            <Cross style={{ width: '24px', height: '24px' }} />
-          </CloseButton>
-        </CloseButtonWrapper>
+export const MessageBox = ({ heading, type, sticky = false, onClose, children, links, t }: Props & WithTranslation) => {
+  const [hideMessageBox, setHideMessageBox] = useState(false);
+  const onCloseMessageBox = () => {
+    setHideMessageBox(true);
+  };
+
+  return (
+    //StickyStyle top:84 makes sure that the messagebox sits beneath the masthead (header ) and the topOffsett sets it so that it applies when reaching the top of the messagebox
+    <Sticky disabled={!sticky} stickyStyle={{ zIndex: 9998, top: 84 }} topOffset={-84}>
+      <Wrapper boxType={type} style={{ display: hideMessageBox ? 'none' : 'flex' }}>
+        <InfoWrapper boxType={type}>
+          <IconWrapper boxType={type}>
+            {type === 'ghost' && <HumanMaleBoard style={{ width: '24px', height: '24px' }} />}
+            {type !== 'ghost' && <InformationOutline style={{ width: '24px', height: '24px' }} />}
+          </IconWrapper>
+          <TextWrapper>
+            {heading && <Label>{heading}</Label>}
+            {children}
+          </TextWrapper>
+        </InfoWrapper>
+        {onClose && (
+          <CloseButtonWrapper>
+            <CloseButton link onClick={onCloseMessageBox}>
+              <Cross style={{ width: '24px', height: '24px' }} aria-hidden={true} />
+            </CloseButton>
+          </CloseButtonWrapper>
+        )}
+      </Wrapper>
+
+      {links && (
+        //loops through the links passed in as properties if there are any and creates a working link for each of them
+        <LinkWrapper style={{ display: hideMessageBox ? 'none' : 'flex' }}>
+          {links.map((x) => (
+            <Link href={x['href']}>{x['name']}</Link>
+          ))}
```

Comment:
Om du typer parameteren slepp du bruke x[''] notasjon.

Reply:
Når jeg typer parameteren får jeg en error som sier "Property 'linkProps' does not exist on type 'never'.".
Det funker derimot med bracket notation.